### PR TITLE
Disable auto-fix `.onmessage` in `prefer-add-event-listener` rule

### LINKED ISSUE #537: Don't auto-fix `.onmessage` in `prefer-add-event-listener` rule
See: https://github.com/xojs/xo/issues/400

### PR DESCRIPTION
**Pull request by fisker**
Fixes #537

### CODE REVIEW

File: rules/prefer-add-event-listener.js [hunk 1]
```diff
@@ -3,14 +3,17 @@ const getDocumentationUrl = require('./utils/get-documentation-url');
 const domEventsJson = require('./utils/dom-events.json');
 
 const message = 'Prefer `{{replacement}}` over `{{method}}`.{{extra}}';
-const beforeUnloadMessage = ' Use `event.preventDefault(); event.returnValue = \'foo\'` to trigger the prompt.';
+const extraMessages = {
+	beforeunload: 'Use `event.preventDefault(); event.returnValue = \'foo\'` to trigger the prompt.',
+	message: 'Notice that there is difference between `SharedWorker#onmessage` and `SharedWorker#addEventListener(\'message\')`.'
```
Comment (sindresorhus) [inline rules/prefer-add-event-listener.js]:
```suggestion
	message: 'Note that there is difference between `SharedWorker#onmessage` and `SharedWorker#addEventListener(\'message\')`.'
```

File: test/prefer-add-event-listener.js [hunk 2]
```diff
@@ -338,6 +343,12 @@ ruleTester.run('prefer-add-event-listener', rule, {
 			`,
 			'onerror',
 			[{excludedPackages: []}]
+		),
+		invalidTestCase(
+			'myWorker.port.onmessage = function(e) {}',
+			null,
```
Comment (sindresorhus) [inline test/prefer-add-event-listener.js]: Use `undefined` over `null`.

Comment (fisker) [reply test/prefer-add-event-listener.js]: I'll update these tests later.

Comment (fisker) [reply test/prefer-add-event-listener.js]: Done

**Comment by sindresorhus** [comment]
Wow. You're are on 🔥 🙌

**Pull request merged**